gym/spaces/dict.py

```diff
@@ -193,6 +202,14 @@ def __repr__(self) -> str:
         """Gives a string representation of this space."""
         return "Dict(" + ", ".join([f"{k}: {s}" for k, s in self.spaces.items()]) + ")"
 
+    def __eq__(self, other) -> bool:
+        """Check whether `other` is equivalent to this instance."""
+        return (
+            isinstance(other, Dict)
+            # Comparison of `OrderedDict`s is order-sensitive
```

Comment:
Why are we not checking the keys are the same? 
Strange that we only checking the spaces

Reply:
Note that `Dict.spaces` is always a `OrderedDict`. `OrderedDict.__eq__` checks both keys and values (implemented `dict.__eq__`). In addition to `dict.__eq__`, the key order is also required by `OrderedDict.__eq__` checks.

Reply:
That makes sense